### PR TITLE
feat: provide a way to create reserved ip independently from vpe creation

### PR DESCRIPTION
**Pull request by jor2**
### Description

https://github.com/terraform-ibm-modules/terraform-ibm-vpe-gateway/issues/395

Suggested approach is to:
1. add an optional input variable to the vpe module that takes a list of reserved ips - eg: reserved_ips
2. extract out the existing logic that reserve ips as a submodule
3. refactor root module to use the new submodule to reserve ips when there is not reserved ips passed in the optional input variable

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- When merging, use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author. The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).
- Merge by using "Squash and merge".

### CODE REVIEW

**Comment by jor2** [comment]
rough draft, will update tomorrow.

**Comment by jor2** [comment]
Need to skip upgrade test

**Comment by ocofaigh** [comment]
Requesting a review from @toddgiguere who has been working on this module

File: main.tf [hunk 1]
```diff
@@ -120,11 +100,11 @@ resource "ibm_is_virtual_endpoint_gateway" "vpe" {
 resource "ibm_is_virtual_endpoint_gateway_ip" "endpoint_gateway_ip" {
   for_each = {
     # Create a map based on endpoint IP
-    for gateway_ip in local.endpoint_ip_list :
+    for gateway_ip in module.create_reserved_ips.endpoint_ip_list :
     (gateway_ip.ip_name) => gateway_ip
   }
   gateway     = local.vpe_map[each.value.gateway_name].id
-  reserved_ip = ibm_is_subnet_reserved_ip.ip[each.key].reserved_ip
+  reserved_ip = var.reserved_ips == null ? module.create_reserved_ips.reserved_ips[each.key].reserved_ip : var.reserved_ips[each.value.ip_name].reserved_ip
```
Comment (toddgiguere) [inline main.tf:107]: Confused about this logic between the for_each of the block and this conditional. For example, if a reserved IP was specified by input variable, why would it also exist in the for_each of this resource block `for gateway_ip in module.create_reserved_ips.endpoint_ip_list`? I'll probably need a description of the overall logic first.

**Review by toddgiguere — CHANGES_REQUESTED** [review]

**Comment by toddgiguere** [comment]
@jor2 I've left some review comments, but overall I'm confused about the new sub module, what it should be used for, and the overall logic of this change. After reading the initial issue and comments I am still unsure what the solution should be, and if this change will solve it.

Let's discuss and I can review further.

**Pull request closed without merging**